### PR TITLE
Change AppCompatActivity to AsyncInitializationActivity

### PR DESCRIPTION
**Pull request by wchen342**
Resolves brave/brave-browser/issues/16642.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java [hunk 1]
```diff
@@ -124,6 +125,11 @@ private void SetDisconnectBtnClickHandler() {
         */
     }
 
+    @Override
+    public boolean shouldStartGpuProcess() {
+        return true;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java:130]: why do we need that?

Comment (wchen342) [reply android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java:130]: The method is abstract. It needs to be implemented in every subclass.

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java:130]: @SergeyZhukovsky I think it's an abstract method. So we need to override ir

Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java:130]: in what cases we return true and false? As I see false in that place https://github.com/brave/brave-core/pull/9256/files#diff-cc8f95891980e64d85e01f6b430c97a504ab9f158e6f20c0e1d1e9a55583809bR75

Comment (wchen342) [reply android/java/org/chromium/chrome/browser/BraveRewardsUserWalletActivity.java:130]: My understanding is if the activity doesn't need to present a GUI then it should be false, as in `SharedClipboardShareActivity.java` and `TrustedVaultKeyRetrievalProxyActivity.java`.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**